### PR TITLE
test: adding some ai tests

### PR DESCRIPTION
**Pull request by mdemin914**
- adding jest + ai.spec
- there were some random ts errors that i just ignored for now
- support for openai for embeddings

### CODE REVIEW

File: src/lib/database/post.ts [hunk 1]
```diff
@@ -407,7 +416,8 @@ export const createPost = async (
         }
       }
     };
-    next(depth);
+    // was this supposed to be awaited?
+    await next(depth);
     return replaceContentWithLinks(post);
```
Comment (johnhenry) [inline src/lib/database/post.ts:421]: // not to be awaited -- this allows for the initial post to be returned while processing the rest of the post. However, this is being overhauled and this code will change.

File: src/lib/database/post.ts [hunk 2]
```diff
@@ -73,7 +73,16 @@ export const getConversation = async (
       }
       currentPost = T;
     }
-    return posts;
+    // many posts to a single target, so gotta fetch them all!
```
Comment (johnhenry) [inline src/lib/database/post.ts:76]: I just want to return a single post and have the other stuff happen in the background.

Comment (johnhenry) [reply src/lib/database/post.ts:76]: I just want to return a single post and have the other stuff happen in the background.

File: src/lib/database/post.ts (hunk 1 above)
Comment (mdemin914) [reply src/lib/database/post.ts:421]: word

File: src/lib/database/post.ts (hunk 2 above)
Comment (mdemin914) [reply src/lib/database/post.ts:76]: not sure i follow? this is used only to generate the agents replies and its already returning an array? how its currently implemented it wont fetch the whole conversation in the case that i have multiple responses to a single post?

Comment (mdemin914) [reply src/lib/database/post.ts:76]: i guess is this a valid use case? https://github.com/johnhenry/antisocial-network/pull/4/files#diff-523f9b2351237923c551363e817ebab21d658a4dd7fc451ee35d6adb12bc8cfeR22

Comment (mdemin914) [reply src/lib/database/post.ts:76]: would the example in the test not be considered a conversation? like i start with a post #1, a bot responds to that post (post #2), and the i add something to post #1 and ask a bot about the whole thing (post #3)? but posts #2 and #3 both would have post #1 as the target. should anything i ask the bot that targets post #1 not include all the related posts?

Comment (johnhenry) [reply src/lib/database/post.ts:76]: A. I think I'm looking at this from last up , and you are looking at it from the the first down:

If you start at #1, you move up to targets, so the conversation would be just [#1]

If you start at #3, then the conversation would be [#3, #1]

If you start at #2, then the conversation would be [#2, #1]

I would consider [#3, #1] and  [#2, #1] to be different conversations. 

Currently there's a "/post merge" command that works similarly to the way you described in that it looks at a post and then looks at all of it's child conversations and creates a merged response. (see lib/database/post.ts/aggregatePostReplies)

(Note, i'm reworking how this operates).

**Comment by johnhenry** [comment]
@mdemin914 thanks for this! I'm closing this PR, but I've pulled the changes into https://github.com/johnhenry/antisocial-network/pull/5

**Pull request closed without merging**